core/src/test/java/org/hibernate/ogm/utils/TestableGridDialect.java

```diff
@@ -27,6 +28,13 @@
 	/**
 	 * Returns the number of entities in the datastore
 	 *
+	 * @param session
+	 */
+	long getNumberOfEntities(Session session);
```

Comment:
Couldn't a new transaction be used for all these assertions with Neo4j Remote? That'd allow sticking to passing the SF as done before.